internal/blockchain/ethereum/quorum/genesis.go

```diff
@@ -0,0 +1,114 @@
+// Copyright © 2024 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package quorum
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+)
+
+type Genesis struct {
+	Config     *GenesisConfig    `json:"config"`
+	Nonce      string            `json:"nonce"`
+	Timestamp  string            `json:"timestamp"`
+	ExtraData  string            `json:"extraData"`
+	GasLimit   string            `json:"gasLimit"`
+	Difficulty string            `json:"difficulty"`
+	MixHash    string            `json:"mixHash"`
+	Coinbase   string            `json:"coinbase"`
+	Alloc      map[string]*Alloc `json:"alloc"`
+	Number     string            `json:"number"`
+	GasUsed    string            `json:"gasUsed"`
+	ParentHash string            `json:"parentHash"`
+}
+
+type GenesisConfig struct {
+	ChainID             int64         `json:"chainId"`
+	HomesteadBlock      int           `json:"homesteadBlock"`
+	Eip150Block         int           `json:"eip150Block"`
+	Eip150Hash          string        `json:"eip150Hash"`
+	Eip155Block         int           `json:"eip155Block"`
+	Eip158Block         int           `json:"eip158Block"`
+	ByzantiumBlock      int           `json:"byzantiumBlock"`
+	ConstantinopleBlock int           `json:"constantinopleBlock"`
+	PetersburgBlock     int           `json:"petersburgBlock"`
+	IstanbulBlock       int           `json:"istanbulBlock"`
+	Clique              *CliqueConfig `json:"clique"`
+}
+
+type CliqueConfig struct {
+	Period int `json:"period"`
+	Epoch  int `json:"epoch"`
+}
+
+type Alloc struct {
+	Balance string `json:"balance"`
+}
+
+func CreateGenesis(addresses []string, blockPeriod int, chainID int64) *Genesis {
+	if blockPeriod == -1 {
+		blockPeriod = 0
```

Comment:
Why did we go for a 0 second block period? If I recall correctly we set 5 as default for Besu

Reply:
This is the same Genesis configurations used for Geth. I kept it the same for consistency. For the purposes of a development environment, 5 seconds can be slightly too long since it slows down transactions and deployments